### PR TITLE
Update optimizers docs page

### PR DESCRIPTION
**Pull request by co9olguy**

### CODE REVIEW

File: doc/introduction/optimizers.rst [hunk 1]
```diff
@@ -6,7 +6,18 @@
...
+Optimizers are objects which can be used to automatically update the parameters of a quantum 
+or hybrid machine learning model. The Optimizers you should use are dependent on your choice
+of classical interface (NumPy, PyTorch, and TensorFlow), and are available from different access
+points. 
+
+Regardless of their origin, all Optimizers provide the same core functionality, 
+and PennyLane is fully compatible with all of them. 
+
+NumPy Interface
+^^^^^^^^^^^^^^^
```
Comment (josh146) [inline doc/introduction/optimizers.rst]: Nice, I like that you separated it into sections! This helps avoid people using PyTorch from attempting to use the built in optimizers.

Side thought: for some of the specialty optimizers (such as QNG, maybe a future Rotoselect), would it be difficult to make them work for PyTorch and TF? I suppose how you do gradients is different enough that we might have to provide TorchQNG, TFQNG, etc.

File: doc/introduction/optimizers.rst [hunk 2]
```diff
@@ -6,7 +6,18 @@
...
+
+When using the standard NumPy interface, PennyLane offers some built-in Optimizers.
```
Comment (josh146) [inline doc/introduction/optimizers.rst]:
```suggestion
When using the standard NumPy interface, PennyLane offers some built-in optimizers.
```

File: doc/introduction/optimizers.rst [hunk 3]
```diff
@@ -6,7 +6,18 @@
 Optimizers
 ==========
 
-When using the standard NumPy interface, PennyLane offers some custom-made optimizers.
+Optimizers are objects which can be used to automatically update the parameters of a quantum 
+or hybrid machine learning model. The Optimizers you should use are dependent on your choice
```
Comment (josh146) [inline doc/introduction/optimizers.rst]:
```suggestion
or hybrid machine learning model. The optimizers you should use are dependent on your choice
```

File: doc/introduction/optimizers.rst [hunk 4]
```diff
@@ -6,7 +6,18 @@
...
+of classical interface (NumPy, PyTorch, and TensorFlow), and are available from different access
+points. 
+
+Regardless of their origin, all Optimizers provide the same core functionality,
```
Comment (josh146) [inline doc/introduction/optimizers.rst]:
```suggestion
Regardless of their origin, all optimizers provide the same core functionality, 
```

File: doc/introduction/optimizers.rst (hunk 3 above)
Comment (josh146) [reply doc/introduction/optimizers.rst]: (because it's not a proper noun I suppose? Looks better in my head this way)

File: doc/introduction/optimizers.rst [hunk 5]
```diff
@@ -24,11 +35,16 @@ Some of these are specific to quantum optimization, such as the :mod:`~.QNGOptim
...
 
-.. warning::
+TensorFlow Interface
+^^^^^^^^^^^^^^^^^^^^
 
-  If using the :ref:`PennyLane PyTorch <torch_interf>`
-  or the :ref:`PennyLane TensorFlow <tf_interf>` interfaces,
-  `PyTorch optimizers <https://pytorch.org/docs/stable/optim.html>`_ and
-  TensorFlow optimizers (found in the module ``tf.train``) should be used respectively.
+When using the :ref:`PennyLane TensorFlow interface<tf_interf>`, you will need to leverage one of
```
Comment (josh146) [inline doc/introduction/optimizers.rst]:
```suggestion
When using the :ref:`PennyLane TensorFlow interface <tf_interf>`, you will need to leverage one of 
```

File: doc/introduction/optimizers.rst (hunk 4 above)
Comment (co9olguy) [inline doc/introduction/optimizers.rst]:
```suggestion
Regardless of their origin, all optimizers provide the same core functionality, 
```

File: doc/introduction/optimizers.rst (hunk 3 above)
Comment (co9olguy) [inline doc/introduction/optimizers.rst]:
```suggestion
or hybrid machine learning model. The optimizers you should use are dependent on your choice
```

File: doc/introduction/optimizers.rst (hunk 2 above)
Comment (co9olguy) [inline doc/introduction/optimizers.rst]:
```suggestion
When using the standard NumPy interface, PennyLane offers some built-in optimizers.
```
